### DISCUSSION TITLE
Can't download GTFS: invalid response received from upstream server

### DISCUSSION THREAD

**Discussion by uq-bfitzpatrick** [transitland/transitland#395] [Q&A] [answered]
Hello,

I can't seem to download zip compressed folders of GTFS data using the feed version endpoint anymore.

I've tried 

` curl https://transit.land/api/v2/rest/feed_versions/a8e502427293c4c1f4f0494c0132c4a47fdcd947/download?apikey=pasted-my-api-key-here`

and visiting the above URL in my browser.

In both cases I get the error message

"An invalid response was received from the upstream server"

I have a 'Hobbyist/Academic' tier API key and was able to download zip compressed folders of GTFS data from Transit.Land last year.

Any help or advice would be most appreciated. Thank you for your time.

**Comment by christophfink** [comment]
I’m facing the exact same issue, tried to use all of the different variable names listed at https://www.transit.land/documentation/index#signing-up-for-an-api-key , both as HTTP header and as URL parameters

--
EDIT: for the record, I tried to access the feed version `963644b0ddb03c54bdaa8af85a90122dd17ebc02`

**Comment by drewda** [comment, marked as answer]
Hi @uq-bfitzpatrick and @christophfink, please email info@interline.io with your account email addresses and/or API keys.

If there's a general bug that needs to be fixed, we'll then post an update here.

---

**Edit**: Never mind, we have been able to reproduce this bug:

It is currently only affecting `/api/v2/rest/feed_versions/{feed_version_key}/download`

`/api/v2/rest/feeds/{feed_key}/download_latest_feed_version` continues to work.

We'll post an update to this thread when we've resolved the issue.

**Reply by drewda** [reply]
We've now deployed a fix for this bug. Both feed version download endpoints should now work again.

//cc @uq-bfitzpatrick @christophfink,

**Reply by uq-bfitzpatrick** [reply]
Marvellous! Thank you for your prompt action!  

I can confirm that 
`/api/v2/rest/feed_versions/{feed_version_key}/download` 
is once again working with my Hobbist/Academic API key (I am particularly interested in downloading historic GTFS data).

**Reply by christophfink** [reply]
Fantastic! I can confirm that it works for me, as well. Thanks for the fast response!

**Reply by drewda** [reply]
Good to hear, and you're welcome.
